File: app/src/androidTest/java/igrek/songbook/dagger/AndroidTestComponent.java
package igrek.songbook.dagger;

import javax.inject.Singleton;

import dagger.Component;
import igrek.songbook.wip.WIPFeatureTest;

@Singleton
@Component(modules = {FactoryModule.class})
public interface AndroidTestComponent extends FactoryComponent {
	
	// enable dagger injection in tests
	void inject(WIPFeatureTest there);
	
}

File: app/src/androidTest/java/igrek/songbook/dagger/AndroidTestComponent.kt
package igrek.songbook.dagger

import dagger.Component
import igrek.songbook.wip.WIPFeatureTest
import javax.inject.Singleton

@Singleton
@Component(modules = [FactoryModule::class])
interface AndroidTestComponent : FactoryComponent {

    // enable dagger injection in tests
    fun inject(there: WIPFeatureTest)

}